File: src/Hacker.java
import java.math.BigInteger;

public class Hacker {
    private String name;

    Hacker() {
        this("Eve");
    }

    Hacker(String name) {
        this.name = name;
    }

    public BigInteger crack(BigInteger c1, BigInteger c2, BigInteger n)
    {
        return c2.subtract(c1).add(new BigInteger("2"))
                .modInverse(n)
                .multiply(c2.add(c1.multiply(new BigInteger("2"))).subtract(new BigInteger("1")))
                .mod(n);
    }


}

File: src/main/java/Hacker.kt
/**
 * Created by John.
 */
import java.math.BigInteger

class Hacker @JvmOverloads internal constructor(private val name: String = "Eve") {

    fun crack(c1: BigInteger, c2: BigInteger, n: BigInteger): BigInteger {
        return ((c2 - c1 + RSA.TWO).modInverse(n) * (c2 + RSA.TWO * c1 - RSA.ONE)) % n
    }
}